tests/unittests/core/test_primary_algo.py

```diff
@@ -65,50 +79,67 @@ def test_init_and_configuration(self, dumbalgo, palgo, fixed_suggestion_value):
             }
         }
 
-    def test_space_can_only_retrieved(self, palgo, space):
+    def test_space_can_only_retrieved(
+        self, palgo: SpaceTransformAlgoWrapper[DumbAlgo], space: Space
+    ):
         """Set space is forbidden, getter works as supposed."""
         assert palgo.space == space
         with pytest.raises(AttributeError):
             palgo.space = 5
 
-    def test_suggest(self, palgo, fixed_suggestion):
+    def test_suggest(
+        self, palgo: SpaceTransformAlgoWrapper[DumbAlgo], fixed_suggestion: Trial
+    ):
         """Suggest wraps suggested."""
         palgo.algorithm.pool_size = 10
-        assert palgo.suggest(1)[0].params == fixed_suggestion.params
-        assert [trial.params for trial in palgo.suggest(4)] == [
-            fixed_suggestion.params
-        ] * 4
-        fixed_suggestion_value = fixed_suggestion
+        trials = palgo.suggest(1)
+        assert trials is not None
+        assert trials[0].params == fixed_suggestion.params
+        ptrials = palgo.suggest(4)
+        # NOTE: Now, if an algorithm has already suggested the same trial, we don't return a
+        # duplicate.
+        assert not ptrials
+        # assert ptrials is not None
```

Comment:
Was this for debugging? Can it be remove or uncommented?

Reply:
Fixed in 44b9a6bf777716b7aca388ee1bfbf7b87d8ec476